### PR TITLE
47 add interact with sc to the standard

### LINKED ISSUE #47: Add interactWithSC to the standard

### PR DESCRIPTION
**Pull request by Elli610**
closes #47

### CODE REVIEW

File: wallet/dapps-communication.md [hunk 1]
```diff
@@ -572,6 +572,118 @@ required:
 
 
 
+</tbody>
+</table>
+
+#### Interact with a smart contract
```
Comment (gregLibert) [inline wallet/dapps-communication.md:578]: Why not call a smart contract ?

File: wallet/dapps-communication.md [hunk 2]
```diff
@@ -572,6 +572,118 @@ required:
...
+<th>Type</th>
+<th>Format</th>
+<th>Example</th>
+</tr>
+</thead>
+<tbody>
+
+<tr>
+<td>Webpage to extension</td>
+<td><code>account.interactWithSC</code></td>
```
Comment (gregLibert) [inline wallet/dapps-communication.md:595]:
```suggestion
<td><code>account.callSC</code></td>
```

File: wallet/dapps-communication.md
Comment (gregLibert) [inline wallet/dapps-communication.md:1]: We should also plan to use read only call. can you have a look at how it is done ? Is this  node API endpoint taking the same parameters as the classic call SC ?
If so, can you add a dry-run argument or equivalent to this function call to make a SC call read only ?

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: wallet/dapps-communication.md
Comment (Elli610) [reply wallet/dapps-communication.md:1]: for the moment, no MassaStation endpoint allows read only calls. I created an [issue](https://github.com/massalabs/thyra/issues/701) about this on the thyra repo.

Comment (gregLibert) [reply wallet/dapps-communication.md:1]: Ok. But if it's possible to do it using the node API, in that case we shall include in the standard even if this is something that is not existing yet on thyra side.

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: wallet/dapps-communication.md [hunk 3]
```diff
@@ -612,12 +612,16 @@ properties:
   amount:
     type: number
     description: Set the amount of MASSA coins given to the block creator.
+  dryRun:
```
Comment (gregLibert) [inline wallet/dapps-communication.md:615]: After reading your definition, I think that this terminology is not clear. I propose to use **Non-Persistent Execution** instead.

File: wallet/dapps-communication.md [hunk 4]
```diff
@@ -612,12 +612,16 @@ properties:
...
+    type: boolean
+    description: If true, the transaction will run in dry-run mode so that the result can be checked without actually broadcasting the transaction.
```
Comment (gregLibert) [inline wallet/dapps-communication.md:617]: The fact that the transaction is not broadcasted is not the main point here: the main purpose is that the modification done to the ledger by the execution of the smart contract is not persisted.

File: wallet/dapps-communication.md [hunk 5]
```diff
@@ -572,6 +572,109 @@ required:
...
+required:
+  - nickname
+  - contractAddress
+  - functionName
+  - parameter
+  - amount
+  - dryRun
```
Comment (gregLibert) [inline wallet/dapps-communication.md:624]: what about having a false default value ?

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: wallet/dapps-communication.md [hunk 6]
```diff
@@ -572,6 +572,119 @@ required:
...
+    type: Args
+    description: Smart contract arguments
+  coins:
+    type: string
+    format: BigInt
+    description: Amount of coins to send to the block creator
+  nonPersistentExecution (optional):
+    type: object
+    properties:
+      isNPE:
```
Comment (gregLibert) [inline wallet/dapps-communication.md:620]: If this field (nonPersistentExecution) is optional, what is the default value of this key ?

File: wallet/dapps-communication.md [hunk 7]
```diff
@@ -572,6 +572,119 @@ required:
...
+<td>Webpage to extension</td>
+<td><code>account.callSC</code></td>
+<td>
+
+```yaml
+type: object
+properties:
+  nickname:
+    type: string
+    description: Account nickname
```
Comment (gregLibert) [inline wallet/dapps-communication.md:603]:
```suggestion
    description: Account nickname to use to sign the operation.
```

File: wallet/dapps-communication.md [hunk 8]
```diff
@@ -572,6 +572,119 @@ required:
...
+  coins:
+    type: string
+    format: BigInt
+    description: Amount of coins to send to the block creator
+  nonPersistentExecution (optional):
+    type: object
+    properties:
+      isNPE:
+        type: boolean
+      maxGas (optional):
```
[the remark this answers is not in the document]
Comment (Elli610) [reply wallet/dapps-communication.md:622]: is "default: false" bellow the "type: boolean" ok?

File: wallet/dapps-communication.md (hunk 6 above)
Comment (Elli610) [reply wallet/dapps-communication.md:620]: is this a duplicate ?

**Review by gregLibert — CHANGES_REQUESTED** [review]

**Review by gregLibert — APPROVED** [review]

**Pull request merged**